Resources/assets_src/src/js/ajax-block.js

```diff
@@ -0,0 +1,26 @@
+/**
+ *
+ * This file is part of the Sonata package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ */
+
+(function ($, global) {
+
+	const getAllBlocks = el => Array.from(el.querySelectorAll('.sonata-dashboard-ajax-block'))
+	const populateBlock = (block, html) => block.innerHTML = html
+	const fetchBlock = block => fetch(block.dataset.url)
+	  .then(response => response.text())
+	  .catch(error => `/!\ Failed to load ${block.dataset.url}`)
+	  .then(html => populateBlock(block, html))
+	const fetchAllBlocks = el => Promise.all(getAllBlocks(el).map(fetchBlock))
+
+	document.addEventListener('DOMContentLoaded', () => fetchAllBlocks(document).then(() => {
+	  // all blocks loaded
+	}))
+
+})(jQuery, window);
```

Comment:
That's very kind of you to blindly trust me! :laughing: 
But the code I showed you was an example off the top of my head, not meant to be copy/pasted.
First, it uses native syntax & features ([ES2015](http://kangax.github.io/compat-table/es6), [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise), [fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API)) that are not supported by all browsers yet - unless [transpiled](https://babeljs.io/repl/#?evaluate=true&lineWrap=true&presets=es2015&experimental=true&loose=false&spec=false&code=const%20getAllBlocks%20%3D%20el%20%3D%3E%20Array.from(el.querySelectorAll('.sonata-dashboard-ajax-block'))%0Aconst%20populateBlock%20%3D%20(block%2C%20html)%20%3D%3E%20block.innerHTML%20%3D%20html%0Aconst%20fetchBlock%20%3D%20block%20%3D%3E%20fetch(block.dataset.url)%0A%20%20.then(response%20%3D%3E%20response.text())%0A%20%20.catch(error%20%3D%3E%20%60%2F!%5C%20Failed%20to%20load%20%24%7Bblock.dataset.url%7D%60)%0A%20%20.then(html%20%3D%3E%20populateBlock(block%2C%20html))%0Aconst%20fetchAllBlocks%20%3D%20el%20%3D%3E%20Promise.all(getAllBlocks(el).map(fetchBlock))%0A%0Adocument.addEventListener('DOMContentLoaded'%2C%20()%20%3D%3E%20fetchAllBlocks(document).then(()%20%3D%3E%20%7B%0A%09%20%20%2F%2F%20all%20blocks%20loaded%0A%7D))) or [polyfilled](https://cdn.polyfill.io/v2/docs/) - (tl;dr: no IE, no Safari).
Second it lacks proper feedback to show the user a request is going on. Check [font-awesome spinners](http://fontawesome.io/examples/#animated) or the AdminLTE's [box component](https://almsaeedstudio.com/themes/AdminLTE/documentation/index.html#component-box) loading states.

Your final code should _look_ like this. :wink: 
```js
(function ($) {

  function fetchBlock ($block) {
    // show a loading indicator
    return $.ajax($block.data('url')).done(function (html) {
      // populate your block with html
    }).fail(function (xhr) {
      // show a nicely formatted error message
    }).always(function () {
      // hide the loading indicator
    })
  }

  $(function () {
    $('.sonata-dashboard-ajax-block').each(function (i, block) {
      var $block = $(block)
      fetchBlock($block)
    })
    // if you want users to be able to refresh the box content
    $(document).on('click', '.sonata-dashboard-ajax-block__refresh-btn', function (event) {
      var $block = $(event.target).closest('.sonata-dashboard-ajax-block')
      fetchBlock($block)
    })
  }) 
}(jQuery)) 
```